core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/UserCoreUtil.java

```diff
@@ -987,8 +987,6 @@ private static boolean isExistingDomain(String domain, int tenantId, DataSource
                 log.debug(errorMessage, e);
             }
             throw new UserStoreException(errorMessage, e);
-        } finally {
-            DatabaseUtil.closeAllConnections(dbConnection, rs, prepStmt);
```

Comment:
We are closing dbConnection  in the upper layer. But don't we need to close result set and prepStmt here.